examples/with-typescript/mixins/auth.ts

```diff
@@ -0,0 +1,62 @@
+import Vue from 'vue'
+import Component from 'nuxt-class-component'
+
+@Component
+export default class AuthMixin extends Vue {
+
+  /**
+   * This define if login got an error
+   * @type {boolean}
+   */
+  loginHasError = false
+
+  /**
+   * Logout method using auth-module with custom post-request
+   * logic, using toast module to show information, success
+   * and error messages.
+   *
+   * @returns {Promise<void>}
+   */
+  async logout () {
+
+    this.$toast.show('Logging out...')
+    await this.$auth.logout().then(() => {
```

Comment:
Doesn't seems logic to me to combine `await` with `then/catch`.
`$toast.success` shoud be called inside a `try` statement after `logout`, and `$toast.error` in the `catch` :
```ts
try {
  await this.$auth.logout()
  this.$toast.success('Successfully disconnected')
} catch (err) {
  this.$toast.error('Error while disconnecting: ' + err.message)
}

Reply:
Agreed, this is copy and pasted from https://github.com/nuxt-community/auth-module/blob/dev/examples/with-class-component/mixins/auth.js, so I'll fix both of them